### PR TITLE
Formula form recipe validation

### PR DESCRIPTION
**Pull request by vrackovic**
- Added validation for `recipe` fields
- All `recipe` fields are grouped in single validation

### CODE REVIEW

File: web/src/main/frontend/src/pages/Formula.js [hunk 1]
```diff
@@ -28,9 +29,18 @@ class Formula extends React.Component {
   };
 
   render() {
-    const categories = this.props.formulaForm
-      .$('recipe')
-      .map(category => <FormulaItem key={category.id} recipe={category} />);
+    let errors = [];
```
Comment (iscekic) [inline web/src/main/frontend/src/pages/Formula.js]:
```js
const errors = {};
```

File: web/src/main/frontend/src/pages/Formula.js [hunk 2]
```diff
@@ -28,9 +29,18 @@ class Formula extends React.Component {
...
+    const categories = this.props.formulaForm.$('recipe').map(category => {
+      if (category.$('categoryId').error) {
+        errors.category = category.$('categoryId').error;
+      }
+      if (category.$('numberOfQuestions').error) {
```
Comment (iscekic) [inline web/src/main/frontend/src/pages/Formula.js]: `\n`

File: web/src/main/frontend/src/pages/Formula.js [hunk 3]
```diff
@@ -28,9 +29,18 @@ class Formula extends React.Component {
...
+        errors.numberOfQuestions = category.$('numberOfQuestions').error;
+      }
+      return (
```
Comment (iscekic) [inline web/src/main/frontend/src/pages/Formula.js]: `\n`

File: web/src/main/frontend/src/pages/Formula.js [hunk 4]
```diff
@@ -66,7 +76,14 @@ class Formula extends React.Component {
                     placeholder="Formula Description"
                     error={this.props.formulaForm.$('description').error}
                   />
-                  <div className="formula__categories_block">{categories}</div>
+                  <div className="formula__categories_block">
+                    {categories}
+                    {errors.category && <FormError field={errors.category} />}
+                    {errors &&
```
Comment (iscekic) [inline web/src/main/frontend/src/pages/Formula.js]: `errors` is always truthy

Comment (vrackovic) [reply web/src/main/frontend/src/pages/Formula.js]: Yup, it is

File: web/src/main/frontend/src/pages/Formula.js [hunk 5]
```diff
@@ -66,7 +78,13 @@ class Formula extends React.Component {
                     placeholder="Formula Description"
                     error={this.props.formulaForm.$('description').error}
                   />
-                  <div className="formula__categories_block">{categories}</div>
+                  <div className="formula__categories_block">
+                    {categories}
+                    {errors.category && <FormError field={errors.category} />}
```
Comment (iscekic) [inline web/src/main/frontend/src/pages/Formula.js]: Let's merge this for now and create an improvement task.

**Pull request merged**